src/mesos_state_client.erl

```diff
@@ -414,7 +414,7 @@ discovery_ports([Port = #{number := Number, protocol := Protocol}|RestPorts],
     PortRecord = #mesos_port{
        name = maps:get(name, Port, undefined),
        number = Number,
-       protocol = protocol(Protocol),
+       protocol = protocol(to_lower(Protocol)),
```

Comment:
If you combine the `protocol()` and `to_lower()` functions then you can save the last `list_to_binary()` conversion